File: leetcode/rust-leetcode/src/443.string-compression.rs
impl Solution {
    pub fn compress(chars: &mut Vec<char>) -> i32 {
        let mut count:i32 = 1;
        let mut s:String = String::new();
        for i in 0..chars.len() {
            if i < chars.len()-1 && chars[i] == chars[i + 1] {
                count += 1;
            } else {
                s.push(chars[i]);
                if count > 1 {
                    for c in count.to_string().chars() {
                        s.push(c);
                    }
                }
                count = 1;
            }
        }
        for (i, c) in s.chars().enumerate() {
            chars[i] = c;
        }
        s.len() as i32
    }
}

File: leetcode/python/443.string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        count = 0
        prev = chars[0]
        i = 0
        for c in chars:
            if c != prev:
                chars[i] = prev
                i += 1
                if count > 1:
                    for rcount in str(count):
                        chars[i] = rcount
                        i += 1
                count = 1
                prev = c
            else:
                count += 1
        chars[i] = prev
        i += 1
        if count > 1:
            for rcount in str(count):
                chars[i] = rcount
                i += 1
        return i

